crypto/bn/bn_lib.c

```diff
@@ -414,24 +415,30 @@ BIGNUM *BN_bin2bn(const unsigned char *s, int len, BIGNUM *ret)
 /* ignore negative */
 static int bn2binpad(const BIGNUM *a, unsigned char *to, int tolen)
 {
-    int i;
+    int i, j, top;
     BN_ULONG l;
 
-    bn_check_top(a);
     i = BN_num_bytes(a);
     if (tolen == -1)
         tolen = i;
     else if (tolen < i)
         return -1;
-    /* Add leading zeroes if necessary */
-    if (tolen > i) {
-        memset(to, 0, tolen - i);
-        to += tolen - i;
+
+    if (i == 0) {
+        OPENSSL_cleanse(to, tolen);
```

Comment:
Well, one can actually argue both ways. Trouble with memset is that you don't know what compiler would do, inline it or call library. And specifically latter can be unproportionally expensive, because it would customarily be done with indirect jump through table. While call to OPENSSL_cleanse would be directly wired even in shared build. Well, as long as you stay with -Bsymbolic, which is the default. That's why I'd customarily use OPENSSL_cleanse with variable length.